libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -630,8 +630,116 @@ string YulUtilFunctions::overflowCheckedIntExpFunction(
 	});
 }
 
+string YulUtilFunctions::overflowCheckedIntLiteralExpFunction(
+	RationalNumberType const& _baseType,
+	IntegerType const& _exponentType,
+	IntegerType const& _commonType
+)
+{
+	solAssert(!_exponentType.isSigned(), "");
+	solAssert(_baseType.isNegative() == _commonType.isSigned(), "");
+	solAssert(_commonType.numBits() == 256, "");
+
+	// Calculates the upperbound for exponentiation, that is, calculate `b`, such that
+	// _base**b <= _maxValue and _base**(b + 1) > _maxValue
+	auto findExponentUpperbound = [](bigint const _base, bigint const _maxValue) -> unsigned
+	{
+		unsigned first = 0;
+		unsigned last = 255;
+		unsigned middle;
+
+		while (first < last)
```

Comment:
Can we somehow reuse an existing binary search implementation?

Reply:
Let me check.

Reply:
A hacky way using `lower_bound` and a custom comparison function would be the following:

```cpp
	auto comp = [&](unsigned a, unsigned) -> bool
	{
		if (
			boost::multiprecision::msb(_base) * a <= boost::multiprecision::msb(_maxValue) &&
			boost::multiprecision::pow(_base, a) <= _maxValue
		)
			return true;
		else
			return false;
	};
  // prev of this index would be what we are looking.
	boost::range::lower_bound(boost::irange(0, 256), 0, comp);
```
I don't find this to be better than explicitly writing the binary search.

Reply:
I'm also fine with the explicit binary search.